<launch>
    <!-- set to false if you don't want to start the kinect -->
    <arg name="kinect" default="true"/>

    <!-- This is a GSC specific config. Set to true to remove the hand from the point cloud, but that will only happen when the full GSC system is running. Set it to false as default and have the GSC launch make it true.-->
    <arg name="filtered" default="false"/>

    <!-- Start the OpenNI camera -->
    <group if="$(arg kinect)">
      <include file="$(find openni_launch)/launch/openni.launch">
        <arg name="depth_registration" value="true" />
      </include>
    </group>

    <!-- start some filters on the point cloud -->
    <group if="$(arg filtered)">
      <include file="$(find sr_point_cloud)/launch/filters.launch" />
    </group>

    <!-- Tracker using NI camera feed as input cloud -->
    <node name="point_cloud_tracker_rgb" pkg="sr_point_cloud" type="tracker_rgb" output="screen">
        <remap from="~input/points" to="/filtered" if="$(arg filtered)" />
      	<remap from="~input/points" to="/camera/depth_registered/points" unless="$(arg filtered)" />
    </node>

    <node name="point_cloud_triangulator" pkg="sr_point_cloud" type="triangulator" output="screen">
        <remap from="~input/points" to="point_cloud_tracker/result/points" />
        <remap from="~output/shape/mesh" to="point_cloud_tracker/result/mesh" />
    </node>
</launch>
